boards/DCM/Inc/App/App_Buzzer.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+struct Buzzer;
+
+/**
+ * Allocate and initialize a buzzer
+ * @param on A function that can be called to turn on the buzzer
+ * @param off A function that can be called to turn off the buzzer
+ * @return A pointer to the created buzzer, whose ownership is given to the
+ *         caller
+ */
+struct Buzzer *App_Buzzer_Create(void (*on)(void), void (*off)(void));
+
+/**
+ * Deallocate the memory used by the given buzzer
+ * @param buzzer The buzzer to deallocate
+ */
+void App_Buzzer_Destroy(struct Buzzer *buzzer);
+
+/**
+ * Turn on the given buzzer
+ * @param buzzer The buzzer to turn on
+ */
+void App_Buzzer_On(const struct Buzzer *buzzer);
```

Comment:
Could we change `On` -> `TurnOn` just to be super clear? With just `On` I think of a check to see if the buzzer is currently on. Similarly for `Off` below.